lib/components/viewers/stop-schedule-table.js

```diff
@@ -0,0 +1,142 @@
+import moment from 'moment'
+import coreUtils from '@opentripplanner/core-utils'
+import React, { Component, createRef } from 'react'
+import styled from 'styled-components'
+
+import Loading from '../narrative/loading'
+import {FETCH_STATUS} from '../../util/constants'
+import {
+  getFirstDepartureFromNow,
+  mergeAndSortStopTimes
+} from '../../util/viewer'
+
+const { formatSecondsAfterMidnight } = coreUtils.time
+
+// Styles for the schedule table and its contents.
+const StyledTable = styled.table`
+  border-spacing: collapse;
+  height: 100%;
+  width: 100%;
+  th {
+    background-color: #fff;
+    box-shadow: 0 1px 0px 0px #ccc;
+    font-size: 75%;
+    position: sticky;
+    top: 0px;
+  }
+  tr > * {
+    border-bottom: 1px solid #ccc;
+    padding: 2px 0 2px 10px;
+    vertical-align: top;
+  }
+  td:first-child, th:first-child {
+    padding-left: 0;
+  }
+`
+
+// Limit the width of the block cell if agency uses very long block IDs,
+// otherwise the table contents overflows the width of the side pane.
+// FIXME: Determine a good way to display long block ids.
+const BlockCell = styled.td`
+  max-width: 100px;
+  overflow-x: hidden;
+  text-overflow: ellipsis;
+`
+const RouteCell = styled.td`
+  font-weight: bold;
+`
+const TimeCell = styled.td`
+  font-weight: bold;
+  text-align: right;
+  white-space: nowrap;
+`
+
+/**
+ * Table showing scheduled departure times for the specified stop organized
+ * chronologically.
+ */
+class StopScheduleTable extends Component {
+  targetDepartureRef = createRef()
+
+  /**
+   * Scroll to the first stop time that is departing from now.
+   */
+  _scrollToFirstDeparture = () => {
+    const { current } = this.targetDepartureRef
+    if (current) {
+      current.scrollIntoView({ behavior: 'smooth', block: 'start' })
+    }
+  }
+
+  componentDidMount () {
+    this._scrollToFirstDeparture()
+  }
+
+  componentDidUpdate () {
+    // Should only happen if user changes date and a new stopData is passed.
+    this._scrollToFirstDeparture()
+  }
+
+  render () {
+    const { date, stopData, timeFormat } = this.props
+    // Show loading spinner if times are still being fetched.
+    if (stopData.fetchStatus === FETCH_STATUS.FETCHING) {
+      return <Loading small />
+    }
+    const mergedStopTimes = mergeAndSortStopTimes(stopData)
+
+    // FIXME: Shift today back one day if the current time is between midnight and the end of the service day.
+    const today = moment().startOf('day').format('YYYY-MM-DD')
+    // Find the next stop time that is departing.
+    // We will scroll to that stop time entry (if showing schedules for today).
+    const shouldHighlightFirstDeparture = mergedStopTimes.length && date === today
+    const highlightedStopTime = shouldHighlightFirstDeparture
+      ? getFirstDepartureFromNow(mergedStopTimes)
+      : null
+
+    return (
+      <StyledTable>
+        <thead>
+          <tr>
+            <th>Block</th>
```

Comment:
Why include block? I don't think this is useful for transit riders.

Reply:
Whether the block column is shown is configurable in https://github.com/opentripplanner/otp-react-redux/pull/292/commits/1c08e0c10a453905bcfcbe1a04763d1828b16506.